README.md

```diff
@@ -32,6 +32,89 @@ flumedb.append({foo: 'bar'}, function (err) {
 })
 ```
 
+## API
+
+### `FlumeviewLevel(version, map) => function`
+
+#### `version`
+The version of the view. Incrementing this number will cause the view to be re-built
+
+#### `map`
+A function with signature `(value, seq)`, where `value` is the item from the log coming past, and `seq` is the location of that value in the flume log.
+
+This function **must return an Array** that's either empty or contains unique index key(s).
+These index keys can then be queired to retrieve the stored value (see `get` and `read` below).
+
+Examples of index key(s) you might return:
+- `[]` - i.e. don't add any indexes for this `value`
+- `['@mix']` - make an index entry for this value under string `@mix`
+- `['@mix', '@mixmix']` - make an index entries for this value under both `@mix` AND `@mixmix`
+- `[['@mix', 1524805117433]]` - make an index entry for this value under the key `['@mix', 1524805117433]` (anything can be a key in leveldb)
+
+This last case is useful when you might want multiple entries under a particular key like `@mix` - if just use `@mix` then the index will get overwritten by future values coming in with the same key.
+Extending the key to include some unique aspect (like a timestamp or the `seq` of the value) means you can have multiple indexes in your view which have a _similar_ key.
+
+e.g. [flumeview-search](https://github.com/flumedb/flumeview-search) is a flumeview which takes the text from incoming values and builds an index which can be searched.
+It takes a sentence like "Learn about leveldb" and maps that into index keys like `[['learn', -145], ['about', -145], ['leveldb', -145]]`, meaning we can later search "leveldb" and find this sentence, as well as all other sentences also containing "leveldb".
```

Comment:
I'm don't get why all three words have "-145."
I am lazy reading and might be missing some important context.

Reply:
it's an arbitrary seq number, it's a hack that flumesearch uses, but agree it looks like magic .. could do with some description...

Reply:
Oh yep.
Is a negative number a valid index?